### ISSUE TITLE
Support plain old javascript models (and also ko-es5 reactive models)

### ISSUE THREAD

**Issue by revengineering** [ErikSchierboom/knockout-pre-rendered#54]
(NOTE: I've already done most of the dev work for this.  Expect a pull request soon.)

I'm using `knockout-pre-rendered` to do server-side-rendering with progressive enhancement.  As a result, my ViewModels are tiny and only have observables for the few data points I actually need in the javascript.

In many cases, I might want properties in my model that aren't "observable" since their values aren't going to change.  As such, it would be great if the `init` binding could write to plain old javascript properties instead of having to use `ko.observable()`.

(Knockout already supports this capability internally, with a few of its "two-way bindings", by generating _property writers_ during the binding preprocessing.)

Using the same techniques within `knockout-pre-rendered`, we can have pure JS models like:

```
// No observables, for people who hate parentheses!
var model = {
    persons: [],
    createPerson: function(firstName, lastName) { 
        return { 
            FirstName: firstName, 
            LastName: lastName,
            showFullName: function() {
                alert(this.FirstName + " " + this.LastName);
            }
        };
    }
};
```
and pre-rendered markup like:
```
<ul data-bind="foreachInit: { data: persons, name: 'personTemplate', createElement: createPerson }">
    <li data-bind="click: showFullName">
        <span data-bind="init, text: FirstName">Bob</span>
        <span data-bind="init, text: LastName">Smith</span>
    </li>
    <li data-bind="click: showFullName">
        <span data-bind="init, text: FirstName">Edward</span>
        <span data-bind="init, text: LastName">Jenkins</span>
    </li>
</ul>
...
```
And the model's `persons[]` array would automatically get populated with two elements, with their `FirstName` and `LastName` properties set correctly.

Nice and clean - and no need to treat properties like functions.

But that's not all! 

This also works with the [knockout-es5 plugin](http://blog.stevensanderson.com/2013/05/20/knockout-es5-a-plugin-to-simplify-your-syntax/), which turns a plain old javascript model into a "reactive" model.

```
// Use ko.track to turn model properties into ES5 getters/setters that wrap observables.
var model = ko.track({
    persons: [],
    createPerson: function() { 
        return ko.track({ 
            FirstName: "", 
            LastName: "",
            showFullName: function() {
                alert(this.FirstName + " " + this.LastName);
            }
        });
    })
};
```
And then you can manipulate the model as plain old javascript and see the changes reflected on the page, just as they would be with observables.
```
model.persons[0].FirstName = "Steve";
model.persons.push(model.createPerson("Phil", "Albertson"));
```
Pull-request forthcoming. Just need to add new tests and some examples.

**Comment by ErikSchierboom** [comment]
@revengineering Wow, this is one great write-up! Looking forward to the PR.

**Comment by revengineering** [comment]
If you're curious and want a preview, feel free to check out my commits in the [POJO-Support branch](https://github.com/revengineering/knockout-pre-rendered/commits/POJO-support).

All existing tests pass. Still need to write new ones for the new code, and add some examples. Might not get to that for a few days though.

**Comment by ErikSchierboom** [comment]
Closed by #55.

@revengineering I've just released version `0.10.0` which includes your awesome contribution. Thanks a lot!

**Comment by revengineering** [comment]
Again, thank you!
